### PR TITLE
Feature/select order in prompt

### PR DESCRIPTION
**Pull request by larisaVasile**
## Description

Pull request for ordering filters in the command bar

Fixes #(number of issue in GitHub)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

## Additional notes:

Add any other notes or comments here.

### CODE REVIEW

File: src/ng-generate/prompter/prompts-questions/shared/prompt-complex-questions.ts [hunk 1]
```diff
@@ -510,3 +510,86 @@ export async function getDatePickerType(templateHelper: TemplateHelper, allAnswe
...
+export async function getCommandBarFilterOrder(templateHelper: TemplateHelper, allAnswers: any, answers: any, aspect: Aspect, options:Schema,enabledCommandBarFunctions:any[]): Promise<object> {
+    return await commandBarFilterOrderPrompt(templateHelper,allAnswers,answers,aspect, options,enabledCommandBarFunctions);
+}
+
+async function orderItems(items:any) {
+    const orderedItems = [...items];
+    let exit = false;
+  
+    while (!exit) {
+      console.log('Order the command bar filters:');
```
Comment (michelu89) [inline src/ng-generate/prompter/prompts-questions/shared/prompt-complex-questions.ts:554]: Remove console.log when is not used for user.

File: src/ng-generate/prompter/prompts-questions/shared/prompt-complex-questions.ts [hunk 2]
```diff
@@ -510,3 +510,86 @@ export async function getDatePickerType(templateHelper: TemplateHelper, allAnswe
...
+      orderedItems.forEach((item, index) => {
+        console.log(`${index + 1}. ${item}`);
```
Comment (michelu89) [inline src/ng-generate/prompter/prompts-questions/shared/prompt-complex-questions.ts:556]: Also this.

File: README.md [hunk 3]
```diff
@@ -244,6 +244,7 @@ the [angular material package](https://v17.material.angular.io/components/table/
     4. Export functionality
     5. Refresh data functionality
     6. Custom actions
+    7. Order filters
```
Comment (michelu89) [inline README.md:247]: "Order of filter"

File: README.md [hunk 4]
```diff
@@ -282,6 +283,7 @@ the [angular material package](https://v17.material.angular.io/components/card/o
...
```
Comment (michelu89) [inline README.md:286]: "Order of filters"

File: src/ng-generate/components/shared/generators/components/command-bar/index.ts [hunk 5]
```diff
@@ -100,3 +92,16 @@ function getPropertiesToCreateFilters(options: any, allProps: Array<Property>):
 function datePickerType(datePickers: Array<DatePicker>, propertyValue: PropValue): string | undefined {
     return datePickers.find((value: any) => value.propertyUrn === propertyValue.propertyUrn)?.datePicker.type;
 }
+
+function sortItemsByArray(orderArray: string[], itemsArray: Property[]): Property[]{
```
Comment (michelu89) [inline src/ng-generate/components/shared/generators/components/command-bar/index.ts:96]: You can also say: 

`function sortItemsByArray(orderArray: string[], itemsArray: Property[]): Property[] {
    const orderMap = orderArray.reduce((map, name, index) => ({ ...map, [name]: index }), {});
    return itemsArray.sort((a, b) => orderMap[a.name] - orderMap[b.name]);
}`

**Comment by sha4be** [comment]
The overall idea of the PR is to add the ability to change filters order at table and card toolbars.
The `setCommandBarFilterOrder` object build algorithm looks the same for tables and cards. The algorithm also requires the same input property `commandbarFunctionalityAnswers.enabledCommandBarFunctions` for `getCommandBarFilterOrder` at shared/prompt-complex-questions. Because of the above I would suggest to move the `setCommandBarFilterOrder` algorithm to `getCommandBarFilterOrder` method and use the single method for table and card promts.

File: src/ng-generate/prompter/prompts-questions/shared/prompt-complex-questions.ts (hunk 1 above)
Comment (larisaVasile) [reply src/ng-generate/prompter/prompts-questions/shared/prompt-complex-questions.ts:554]: done

File: README.md (hunk 3 above)
Comment (larisaVasile) [reply README.md:247]: done

File: README.md (hunk 4 above)
Comment (larisaVasile) [reply README.md:286]: done

File: src/ng-generate/components/shared/generators/components/command-bar/index.ts (hunk 5 above)
Comment (larisaVasile) [reply src/ng-generate/components/shared/generators/components/command-bar/index.ts:96]: done

File: src/ng-generate/prompter/prompts-questions/shared/prompt-complex-questions.ts (hunk 2 above)
Comment (larisaVasile) [reply src/ng-generate/prompter/prompts-questions/shared/prompt-complex-questions.ts:556]: i need this, it is used when the filter properties are shown

**Review by sha4be — APPROVED** [review]

**Pull request merged**